### ISSUE TITLE
Editor - Should allow configuration of encoding behavior.

### ISSUE THREAD

**Issue by colin-jack** [jpuri/react-draft-wysiwyg#269]
I realize the editor is decoupled from the format (HTML/Markdown) it produces. However I think it would be useful if it could be configured to perform the appropriate encoding based on the format being produced.

As an example if I am outputting Markdown and the user types in "    Bob" then I would expect the editor to encode the white-space, rather than passing it as-is resulting in the creation of a unexpected code block.

**Comment by jpuri** [comment]
Hey @colin-jack:

Its not markdown editor and that is not even the intention - its rich text editor and mark-down is just an output format that is supported.

I hope I correctly understood your issue.

**Comment by colin-jack** [comment]
Hi @jpuri:

That makes sense, I will follow up with the issue with draft-js-export-markdown. 

However I wondered if you had also considered allowing registering a handler for [handleBeforeInput](https://draftjs.org/docs/api-reference-editor.html#handlebeforeinput ) as it seems like this would allow us to sanitise the input before it is processed?

**Comment by jpuri** [comment]
Good idea I should add handleBeforeInput and other useful callbacks as props, I will try to soon prioritize that.

**Comment by colin-jack** [comment]
That's brilliant and thanks for the quick reply.
